web/src/components/organisms/Workspace/Settings/hooks.ts

```diff
@@ -0,0 +1,61 @@
+import { useCallback } from "react";
+import { useNavigate } from "react-router-dom";
+
+import Notification from "@reearth-cms/components/atoms/Notification";
+import {
+  useUpdateWorkspaceMutation,
+  useDeleteWorkspaceMutation,
+} from "@reearth-cms/gql/graphql-client-api";
+import { useT } from "@reearth-cms/i18n";
+import { useWorkspace } from "@reearth-cms/state";
+
+export default () => {
+  const navigate = useNavigate();
+  const [currentWorkspace, setCurrentWorkspace] = useWorkspace();
+  const t = useT();
+
+  const workspaceId = currentWorkspace?.id;
+  const workspaceName = currentWorkspace?.name;
+
+  const [updateWorkspaceMutation] = useUpdateWorkspaceMutation();
+  const [deleteWorkspaceMutation] = useDeleteWorkspaceMutation({
+    refetchQueries: ["GetMe"],
+  });
+
+  const handleWorkspaceUpdate = useCallback(
+    async (name?: string) => {
+      if (!workspaceId || !name) return;
+      const workspace = await updateWorkspaceMutation({
```

Comment:
`const resp` or `res` is more accurate and anyways common.

Reply:
But workspace isn't what is represented here. This is a response object with various fields.